raftLogCheck.js

```diff
@@ -0,0 +1,235 @@
+                }
+                log.info('getting member page', { member, seq });
+                log.debug('memberLog:', { memberLog });
+                let i;
+                let j;
+                for (i = 0;
+                     i < pageSize;
+                     i++) {
+                    if (leaderLog.log !== undefined &&
+                        leaderLog.log[i] !== undefined &&
+                        leaderLog.log[i].entries !== undefined) {
+                        for (j = 0;
+                             j < leaderLog.log[i].entries.length;
+                             j++) {
+                            let expectKey = true;
+                            let expectValue = true;
+                            if (memberLog.log === undefined ||
+                                memberLog.log[i] === undefined ||
+                                memberLog.log[i].entries ===
+                                undefined) {
+                                log.warning('undefined entries at',
+                                            { seq: seq + i, entry: j });
+                                continue;
+                            }
+                            if (leaderLog.log[i].entries[j].key === undefined) {
+                                if (leaderLog.log[i].method === 0 || // CREATE
+                                    leaderLog.log[i].method === 1 || // DELETE
+                                    leaderLog.log[i].method === 7) { // PUT_ATTR
+                                    expectKey = false;
+                                } else {
+                                    log.warning('undefined leader key at', {
+                                        seq: seq + i,
+                                        off: j,
+                                        db: leaderLog.log[i].db,
+                                        entry: leaderLog.log[i].entries[j] });
+                                    continue;
+                                }
+                            }
+                            if (leaderLog.log[i].entries[j].value ===
+                                undefined) {
+                                if (leaderLog.log[i].entries[j].type ===
+                                    'del') {
+                                    expectValue = false;
+                                } else {
+                                    log.debug('undefined leader value at', {
+                                        seq: seq + i,
+                                        off: j,
+                                        entry: leaderLog.log[i].entries[j] });
+                                    continue;
+                                }
+                            }
+                            if (memberLog.log[i].entries[j].key ===
+                                undefined) {
+                                if (expectKey) {
+                                    log.debug('undefined member key at', {
```

Comment:
Is it supposed to be a warning log, not debug? Same question for the following debug logs